core/src/commonMain/kotlin/com/xebia/functional/xef/llm/MemoryManagement.kt

```diff
@@ -4,85 +4,51 @@ import com.xebia.functional.xef.conversation.Conversation
 import com.xebia.functional.xef.llm.models.chat.Choice
 import com.xebia.functional.xef.llm.models.chat.ChoiceWithFunctions
 import com.xebia.functional.xef.llm.models.chat.Message
-import com.xebia.functional.xef.llm.models.chat.Role
 import com.xebia.functional.xef.store.ConversationId
 import com.xebia.functional.xef.store.Memory
-import io.ktor.util.date.*
 
-internal suspend fun List<Message>.addToMemory(chat: LLM, scope: Conversation) {
-  val memories = toMemory(chat, scope)
-  if (memories.isNotEmpty()) {
-    scope.store.addMemories(memories)
+internal suspend fun List<Message>.addToMemory(scope: Conversation) {
+  val cid = scope.conversationId
+  if (cid != null) {
+    val memories = toMemory(scope)
+    if (memories.isNotEmpty()) {
+      scope.store.addMemories(memories)
+    }
   }
 }
 
-internal fun Message.toMemory(cid: ConversationId, chat: LLM, delta: Int = 0): Memory =
-  Memory(
-    conversationId = cid,
-    content = this,
-    timestamp = getTimeMillis() + delta,
-    approxTokens = chat.tokensFromMessages(listOf(this))
-  )
+internal fun Message.toMemory(cid: ConversationId, index: Int): Memory =
+  Memory(conversationId = cid, content = this, index = index)
 
-internal fun List<Message>.toMemory(chat: LLM, scope: Conversation): List<Memory> {
+internal fun List<Message>.toMemory(scope: Conversation): List<Memory> {
   val cid = scope.conversationId
   return if (cid != null) {
-    mapIndexed { index, it -> it.toMemory(cid, chat, index) }
+    mapIndexed { index, it -> it.toMemory(cid, scope.store.incrementIndexAndGet()) }
```

Comment:
I think we are not using the index, right? Could we change it to `map`?

Reply:
Good catch